core/services/ocr2/plugins/ocr2keeper/evm21/core/type.go

```diff
@@ -26,36 +15,23 @@ const (
 	upkeepTypeByteIndex = 15
 )
 
-// getUpkeepType returns the upkeep type from the given ID.
+// GetUpkeepType returns the upkeep type from the given ID.
 // it follows the same logic as the contract, but performs it locally.
 //
 // TODO: check endianness
-func getUpkeepType(id ocr2keepers.UpkeepIdentifier) upkeepType {
+func GetUpkeepType(id ocr2keepers.UpkeepIdentifier) ocr2keepers.UpkeepType {
```

Comment:
The issue is that `UpkeepIdentifier` is defined in the plugin repo, while the type might be chain dependent and therefore this will always come from the core node.

In addition, function interfaces fits well in cases like that, where we want to expose pure functions.

Reply:
Discussed offline. Keeping this function in core as this might be chain dependent for now